integration-tests/support/pages/BuildApplicationWithGitSourcePage.ts

```diff
@@ -0,0 +1,43 @@
+import { createApplicationPagePO } from '../pageObjects/createApplication-po';
+import { buildApplicationWithGitSourcePagePO } from '../pageObjects/createApplication-po';
+import { AbstractWizardPage } from './AbstractWizardPage';
+export class BuildApplicationWithGitSourcePage extends AbstractWizardPage {
+  setSource(source: string) {
+    this.clearSourceField();
+    cy.get(buildApplicationWithGitSourcePagePO.enterSource).type(source);
+  }
+
+  clearSourceField() {
+    cy.get(buildApplicationWithGitSourcePagePO.enterSource).clear();
+  }
+
+  isValidated() {
+      cy.get(buildApplicationWithGitSourcePagePO.validated).should('have.text', 'Validated')
+  }
+
+  clickGitOptions() {
+    cy.contains(buildApplicationWithGitSourcePagePO.gitOptions).click();
+  }
+
+  setGitReference(gitReference: string) {
+    this.clearGitReference();
+    cy.get(buildApplicationWithGitSourcePagePO.gitReference).type(gitReference);
+  }
+
+  clearGitReference() {
```

Comment:
I'm not sure if we need this logic to be separated into an extra method as it contains the reference usable only with this particular element. Applies to other "clear" methods too.

Reply:
Looks better to me, thanks. :slightly_smiling_face: